### ISSUE TITLE
Loading invalid image messes up GUI

### ISSUE THREAD

**Issue by aljelly** [Philip-Scott/Spice-up#170]
It's probably very unlikely that someone will try to insert an invalid image into a presentation, but still:

### Steps to reproduce
1. Download [this invalid image](https://user-images.githubusercontent.com/10395308/32141515-69f2ff42-bc7a-11e7-950e-f8cf3db70a42.png).
2. Insert it into Spice-Up.

### Expected result
Either:
- The image is loaded but shows with some placeholder icon indicating it couldn't be read, like a generic image icon or invalid icon
- Or an error message informing me that the image couldn't be loaded

### Actual result
Most of the GUI turns into solid colors. The terminal outputs errors (see below).

![2017-10-30 09_11_22-elementary os running - oracle vm virtualbox](https://user-images.githubusercontent.com/10395308/32162938-8267fc84-bd52-11e7-9131-b37d575535d0.png)

Some GUI elements come back as they're interacted with (like buttons, the presentation thumbnail), but the above terminal output repeats when you do this and the GUI is still mostly messed up. The elements that come back when you interact with them disappear again if you resize the window.

Once you find the invalid image in the slide and remove it, or remove the slide the invalid image is in, the GUI is able to fix itself after doing something like resizing the window.

### Terminal output

```
$ com.github.philip-scott.spice-up 
[INFO 09:06:10.921739] Application.vala:153: Spice-Up version: 0.8
[INFO 09:06:10.921793] Application.vala:155: Kernel version: 4.10.0-37-generic
[WARNING 09:07:12.649692] [Gtk] Error loading image 'file:///tmp/spice-up-aled-img-0.png': Unrecognized image file format
[WARNING 09:07:12.659338] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:12.661275] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:12.662590] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:12.662676] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:12.665433] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:12.665548] [Gtk] drawing failure for widget 'SpiceCanvas': invalid matrix (not invertible)
[WARNING 09:07:12.665588] [Gtk] drawing failure for widget 'GtkStack': invalid matrix (not invertible)
[WARNING 09:07:12.665622] [Gtk] drawing failure for widget 'GtkAspectFrame': invalid matrix (not invertible)
[WARNING 09:07:12.665739] [Gtk] drawing failure for widget 'GtkGrid': invalid matrix (not invertible)
[WARNING 09:07:12.665773] [Gtk] drawing failure for widget 'GtkStack': invalid matrix (not invertible)
[WARNING 09:07:12.665805] [Gtk] drawing failure for widget 'GtkOverlay': invalid matrix (not invertible)
[WARNING 09:07:12.665842] [Gtk] drawing failure for widget 'SpiceWindow': invalid matrix (not invertible)
[WARNING 09:07:12.970953] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:12.972609] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:12.974075] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:12.974293] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:12.975578] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:12.975767] [Gtk] drawing failure for widget 'SpiceCanvas': invalid matrix (not invertible)
[WARNING 09:07:12.975871] [Gtk] drawing failure for widget 'GtkStack': invalid matrix (not invertible)
[WARNING 09:07:12.975976] [Gtk] drawing failure for widget 'GtkAspectFrame': invalid matrix (not invertible)
[WARNING 09:07:12.976128] [Gtk] drawing failure for widget 'GtkGrid': invalid matrix (not invertible)
[WARNING 09:07:12.976232] [Gtk] drawing failure for widget 'GtkStack': invalid matrix (not invertible)
[WARNING 09:07:12.976335] [Gtk] drawing failure for widget 'GtkOverlay': invalid matrix (not invertible)
[WARNING 09:07:12.976442] [Gtk] drawing failure for widget 'SpiceWindow': invalid matrix (not invertible)
[WARNING 09:07:13.811927] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:13.819371] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:13.820870] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:13.820972] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:13.822182] [Gtk] drawing failure for widget 'SpiceImageItem': invalid matrix (not invertible)
[WARNING 09:07:13.822256] [Gtk] drawing failure for widget 'SpiceCanvas': invalid matrix (not invertible)
[WARNING 09:07:13.822293] [Gtk] drawing failure for widget 'GtkStack': invalid matrix (not invertible)
[WARNING 09:07:13.822330] [Gtk] drawing failure for widget 'GtkAspectFrame': invalid matrix (not invertible)
[WARNING 09:07:13.822370] [Gtk] drawing failure for widget 'GtkGrid': invalid matrix (not invertible)
[WARNING 09:07:13.822406] [Gtk] drawing failure for widget 'GtkStack': invalid matrix (not invertible)
[WARNING 09:07:13.822442] [Gtk] drawing failure for widget 'GtkOverlay': invalid matrix (not invertible)
[WARNING 09:07:13.822483] [Gtk] drawing failure for widget 'SpiceWindow': invalid matrix (not invertible)
```

**Comment by Philip-Scott** [comment]
Looks like i am catching the error `[WARNING 09:07:12.649692] [Gtk] Error loading image 'file:///tmp/spice-up-aled-img-0.png': Unrecognized image file format` but basically just didn't do anything with it :sweat_smile: 

Thank you for bringing it up!

**Comment by Philip-Scott** [comment]
Fixed on facb55f :) 

Thank you for bringing this up!
